### PR TITLE
Add list-modules cmd

### PR DESCRIPTION
**Pull request by msoedov**
#60

### CODE REVIEW

File: monkeytype/cli.py [hunk 1]
```diff
@@ -168,6 +168,16 @@ def print_stub_handler(args: argparse.Namespace, stdout: IO, stderr: IO) -> None
     print(output, file=file)
 
 
+def print_list_handler(args: argparse.Namespace, stdout: IO, stderr: IO) -> None:
```
Comment (mpage) [inline monkeytype/cli.py]: Can you rename this to `list_modules_handler` so that it more closely matches the command that it implements?

**Comment by carljm** [comment]
Yeah, I agree with the NotImplementedError plan. We can’t break existing functionality with existing third party backends. It’s ok if the new functionality requires the backend to be extended before it works though. The change log entry should discuss this clearly to avoid confusion from the NotImplementedError.

**Comment by msoedov** [comment]
Good point, I missed to update `CallTraceStore` interface

**Comment by msoedov** [comment]
I have been also thinking about another flag `monkeytype apply_all` or `monkeytype apply *`. However with the current patch it will be possible to achieve the same  results via`monkeytype list-modules | xargs -L monkeytype apply`. WDYT?

**Comment by johnarnold** [comment]
monkeytype apply * is getting really close to a regex implementation... That would be pretty clever and more flexible than just "all", IMO.  Although you could accomplish the same thing with "monkeytype list_all | grep foo_module | xargs monkeytype apply",  I think supporting a regex and a loop internally would probably be faster and certainly more discoverable / easy to document.

**Comment by carljm** [comment]
We've already gotten this feature request once, it's tracked in #26. Personally I still favor a `--recursive` flag; I think this covers the common use cases and is simpler to document and implement. For more advanced use cases you can always pipe (maybe filtered) `list_modules` output to monkeytype (and I don't mind documenting that). I'm not very excited about building module matching by regex into MonkeyType.

**Comment by johnarnold** [comment]
@carljm I'm fine with pipe-based filtering and --recursive too. Just thinking out loud :)

**Comment by msoedov** [comment]
I did not mean using regexp in example  `monkeytype apply *`. `*` is just wildcard value which is kind of apply patch to all modules.  I think even more intuitive would be use  `monkeytype apply .` . Flags as `-r` / `--recursive` is a good option but `monkeytype apply .`  `monkeytype stub .` is easier to remember

**Comment by carljm** [comment]
Looks great! Thanks for doing this, merging. We can continue the other discussion on #26.

**Comment by msoedov** [comment]
My pleasure

**Pull request merged**